payone-adapter/src/main/java/com/commercetools/payment/payone/methods/PayoneCreatePaymentMethodBase.java

```diff
@@ -65,4 +72,21 @@ protected static String getLanguageFromPaymentDataOrFallback(CreatePaymentData c
                         .map(CreatePaymentMethodBase::getLanguageFromCart)
                         .orElseGet(Locale.ENGLISH::getLanguage));
     }
+
+    @Override
+    public Function<CreatePaymentData, CompletionStage<PaymentCreationResult>> create() {
+        return cpd ->
+                addNewPayment(cpd)
+                    .thenApply(payment -> null != payment
+                        ? PaymentCreationResultBuilder
+                                .of(OperationResult.SUCCESS)
+                                .payment(payment)
+                                 // so far for the payments we use only CONTINUE handling task action,
+                                 // aka "proceed to checkout page",
+                                 // but later a transactions creation results could be REDIRECT (like for paypal)
+                                .handlingTask(HandlingTask.of(ShopAction.CONTINUE))
+                                .build()
+                        : PaymentCreationResultBuilder
+                            .ofError(format("An error occurred during creation of %s payment object.", getMethodType())));
```

Comment:
looks like you are right: instead of that we should have `exceptionally()` catcher